server/src/main/java/io/littlehorse/server/KafkaStreamsServerImpl.java

```diff
@@ -241,7 +241,7 @@ private RequestExecutionContext requestContext() {
     public KafkaStreamsServerImpl(LHServerConfig config) {
         MetadataCache metadataCache = new MetadataCache();
         this.config = config;
-        this.taskQueueManager = new TaskQueueManager(this);
+        this.taskQueueManager = new TaskQueueManager(this, 20_000);
```

Comment:
I just realized that 20,000 might be a bit too big. Perhaps 1,000. Because we could easily have 1,000 different taskDef's, which means 20M total tasks...yikes.

This is a similar problem to the WriteBufferManager deciding when to flush Memtables in Speedb :wink: